### PR TITLE
Add Register3D _to_dict method and UTs

### PR DESCRIPTION
**Pull request by Codoscope**
#299

### CODE REVIEW

File: pulser/register.py [hunk 1]
```diff
@@ -282,6 +282,29 @@ def _draw_checks(
                     "Needs more than one atom to draw " "the blockade radius."
                 )
 
+    @abstractmethod
```
Comment (HGSilveri) [inline pulser/register.py:285]: What's the motivation to define it as an `abstractmethod` that is the same on all subclasses? Is it to force the definition of `_to_dict()` on each subclass (even if redundant) so it requires coverage by the unit tests?

File: pulser/register.py [hunk 2]
```diff
@@ -282,6 +282,29 @@ def _draw_checks(
...
+    def _to_dict(self) -> dict[str, Any]:
+        qs = dict(zip(self._ids, map(np.ndarray.tolist, self._coords)))
+        return obj_to_dict(self, qs)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, BaseRegister):
```
Comment (HGSilveri) [inline pulser/register.py:291]: Isn't this unnecessary considering you then check for `type(self) is type(other)` ?

File: pulser/register.py [hunk 3]
```diff
@@ -282,6 +282,29 @@ def _draw_checks(
...
+            type(self) is type(other)
+            and set(self._ids) == set(other._ids)
+            and all(
+                (
+                    np.array_equal(
+                        self._coords[i],
+                        other._coords[other._ids.index(self._ids[i])],
+                    )
+                    for i in range(len(self._ids))
+                )
```
Comment (HGSilveri) [inline pulser/register.py:304]: I would loop over the `_ids` instead:
```suggestion
                (
                    np.array_equal(
                        self._coords[i],
                        other._coords[other._ids.index(id)],
                    )
                    for i, id in enumerate(self._ids)
                )
```

File: pulser/sequence.py [hunk 4]
```diff
@@ -215,6 +215,11 @@ def qubit_info(self) -> dict[QubitId, np.ndarray]:
         """Dictionary with the qubit's IDs and positions."""
         return self._register.qubits
 
+    @property
+    def register(self) -> BaseRegister:
+        """Register with the qubit's IDs and positions."""
+        return self._register
```
Comment (HGSilveri) [inline pulser/sequence.py:221]: I agree with this addition. I think that `Sequence._register` is being accessed in `_seq_drawer.py`, do you think you could also replace those calls with `Sequence.register`?

**Review by HGSilveri — COMMENTED** [review]
Looking good! I only have a few remarks and requests

File: pulser/register.py (hunk 1 above)
Comment (Codoscope) [reply pulser/register.py:285]: The motivation is to make sure that any future subclass won't be wrongly serializable by default, and will need to make it explicit by adding a new `_to_dict()` method. The developer will also be forced to think about whether they want the method to be identical, or whether they want to extend it.

Comment (HGSilveri) [reply pulser/register.py:285]: Yes, that's a fair point.

**Review by HGSilveri — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**